src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java

```diff
@@ -55,19 +57,21 @@ public FormMetadataPort syncWithFilesAt(Path storageRoot) {
     Stream<Path> formFiles = candidateFormFiles.filter(path -> isAForm(XmlElement.from(path)));
 
     // Parse existing metadata.json files or build new FormMetadata from form files
-    Stream<FormMetadata> metadataFiles = formFiles.map(formFile -> {
+    // At this point, we collect the stream to avoid problems coming
+    // from actually writing files in the same folders we're reading from
```

Comment:
What sort of problems?

Reply:
Classic reading stuff that you're writing to. The adapter scans, reads and writes the same files in the same stream.